### PR TITLE
Convert csh script distscript.csh into a POSIX sh script

### PR DESCRIPTION
**Pull request by hjelmn**
This commit removes the only cshell script in the Open MPI
project.

Signed-off-by: Nathan Hjelm <hjelmn@lanl.gov>

### CODE REVIEW

**Comment by hjelmn** [comment]
@jsquyres Please let me know what you think.

**Comment by hjelmn** [comment]
Also part of this commit: allow use of curl if wget is not available (OSX does not ship with wget)

**Comment by mellanox-github** [comment]
* Coverity found 905 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_474/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_474/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_474/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/474/

**Build Log**
*last 50 lines*

 ```
[...truncated 33418 lines...]
+ for OMPI_HOME in '$(echo $ompi_home_list)'
+ echo 'check if mca_base_env_list parameter is supported in /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1'
check if mca_base_env_list parameter is supported in /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1
++ /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/bin/ompi_info --param mca base --level 9
++ grep mca_base_env_list
++ wc -l
+ val=2
+ '[' 2 -gt 0 ']'
+ echo 'test mca_base_env_list option in /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1'
test mca_base_env_list option in /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1
+ export XXX_C=3 XXX_D=4 XXX_E=5
+ XXX_C=3
+ XXX_D=4
+ XXX_E=5
++ /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/bin/mpirun -np 2 -mca mca_base_env_list 'XXX_A=1;XXX_B=2;XXX_C;XXX_D;XXX_E' env
++ grep '^XXX_'
++ wc -l
+ val=10
+ '[' 10 -ne 10 ']'
+ echo 'mca_base_env_list=XXX_A=1;XXX_B=2;XXX_C;XXX_D;XXX_E'
++ grep '^XXX_'
++ /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/bin/mpirun -np 2 -am /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/test_amca.conf /scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/jenkins_scripts/jenkins/ompi/env_mpi
++ wc -l
+ val=0
+ '[' 0 -ne 10 ']'
+ exit 1
Build step 'Execute shell' marked build as failure
TAP Reports Processing: START
Looking for TAP results report in workspace using pattern: **/*.tap
Saving reports...
Processing '/var/lib/jenkins/jobs/gh-ompi-master-pr/builds/474/tap-master-files/cov_stat.tap'
Parsing TAP test result [/var/lib/jenkins/jobs/gh-ompi-master-pr/builds/474/tap-master-files/cov_stat.tap].
not ok - coverity detected 905 failures in all_474 # SKIP http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_474/output/errors/index.html
not ok - coverity detected 5 failures in oshmem_474 # TODO http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_474/output/errors/index.html
ok - coverity found no issues for yalla_474
ok - coverity found no issues for mxm_474
not ok - coverity detected 2 failures in fca_474 # TODO http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_474/output/errors/index.html
ok - coverity found no issues for hcoll_474

TAP Reports Processing: FINISH
coverity_for_all	http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_474/output/errors/index.html
coverity_for_oshmem	http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_474/output/errors/index.html
coverity_for_fca	http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_474/output/errors/index.html
[copy-to-slave] The build is taking place on the master node, no copy back to the master will take place.
Setting commit status on GitHub for https://github.com/open-mpi/ompi/commit/0d8eae39fd5af792d8939a9047dbb75437c08037
[BFA] Scanning build for known causes...
[BFA] No failure causes found
[BFA] Done. 0s
Setting status of b1ea28d4697a838029d26dba5b81dceca0489d6d to FAILURE with url http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr/474/ and message: Merged build finished.
```

Test FAILed.

**Comment by hjelmn** [comment]
another random jenkins failure

**Comment by hjelmn** [comment]
bot:retest

**Comment by mellanox-github** [comment]
* Coverity found 905 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_475/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_475/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_475/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/475/
Test PASSed.

File: config/distscript.sh [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+# Copyright (c) 2015      Los Alamos National Security, LLC. All rights
+#                         reserved.
+# $COPYRIGHT$
+#
+# Additional copyrights may follow
+#
+# $HEADER$
+#
+
+srcdir="$1"
```
Comment (jsquyres) [inline config/distscript.sh]: As long as you're doing this, you might as well remove the unnecessary quotes on srcdir, OMPI_VERSION, and OMPI_REPO_REV.  :-)

File: config/distscript.sh [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+#########################################################
+# VERY IMPORTANT: Now go into the new distribution tree #
+#########################################################
+cd "$distdir"
+echo "*** Now in distdir: $distdir"
+
+#
+# Get the latest config.guess and config.sub from ftp.gnu.org
+#
```
Comment (jsquyres) [inline config/distscript.sh]: Where did you get this config.guess/config.sub stuff from?  We don't do this on master any more -- did you start with a v1.8 version of distscript.csh?

None of this code should be on master any more (lines 86 through 167)

**Comment by mellanox-github** [comment]
* Coverity found 905 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_476/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_476/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_476/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/476/
Test PASSed.

File: config/distscript.sh [hunk 3]
```diff
@@ -20,30 +22,30 @@
...
-set srcdir="$1"
-set builddir="`pwd`"
-set distdir="$builddir/$2"
-set OMPI_VERSION="$3"
-set OMPI_REPO_REV="$4"
+srcdir="$1"
+builddir=$PWD
+distdir="$builddir/$2"
+OMPI_VERSION="$3"
+OMPI_REPO_REV="$4"
```
Comment (jsquyres) [inline config/distscript.sh]: Are you going to remove these useless quotes, too?  :-)

Comment (jsquyres) [inline config/distscript.sh]: See my comment on your original diff: https://github.com/open-mpi/ompi/pull/552#discussion_r28983516

**Comment by mellanox-github** [comment]
* Coverity found 905 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_477/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_477/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_477/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/477/
Test PASSed.

File: config/distscript.sh (hunk 3 above)
Comment (hjelmn) [inline config/distscript.sh]: Yup. Going to make another update as soon as I finish with osc/pt2pt.

**Comment by hjelmn** [comment]
Any csh script I consider a bug

**Comment by hjelmn** [comment]
Took out the unnecessary quoting. Should be good to go.

**Comment by mellanox-github** [comment]
* Coverity found 905 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_480/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_480/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_480/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/480/
Test PASSed.

**Comment by jsquyres** [comment]
:+1:

(when I told Nathan that he was free to re-write the csh script and submit a PR, I didn't think he'd really do it... :-) )

**Pull request merged**